### PR TITLE
Rename diagram-viewer into network-viewer

### PR DESCRIPTION
**Pull request by flo-dup**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**
Renaming

**What is the current behavior?**
`powsybl-diagram-viewer` / `@powsybl/diagram-viewer`

**What is the new behavior (if this is a feature change)?**
`powsybl-network-viewer` / `@powsybl/network-viewer`

**Does this PR introduce a breaking change or deprecate an API?**
- [x] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

'npm install @powsybl/network-viewer' instead of 'npm install @powsybl/diagram-viewer'

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
-# powsybl-diagram-viewer
+# powsybl-network-viewer
 
-Typescript library to integrate a powsybl svg diagram in a javascript project. The library is built with the Vite bundler.
+Typescript library to integrate network viewers in a javascript project.
+This library contains three viewers: 
+- single-line-diagram-viewer, to integrate the SVG representing a graph of voltage levels within a network, the SVG and the corresponding metadata JSON file being generated by [powsybl-diagram](https://github.com/powsybl/powsybl-diagram)
```
Comment (etiennehomer) [inline README.md:5]:
```suggestion
- network-area-diagram-viewer, to integrate the SVG representing a graph of voltage levels within a network, the SVG and the corresponding metadata JSON file being generated by [powsybl-diagram](https://github.com/powsybl/powsybl-diagram)
```

File: README.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+- network-area-diagram-viewer, to integrate the SVG representing single-line diagrams, the SVG and the corresponding metadata JSON file being generated by  [powsybl-diagram](https://github.com/powsybl/powsybl-diagram)
```
Comment (etiennehomer) [inline README.md:6]:
```suggestion
- single-line-diagram-viewer, to integrate the SVG representing single-line diagrams, the SVG and the corresponding metadata JSON file being generated by  [powsybl-diagram](https://github.com/powsybl/powsybl-diagram)
```

File: package.json [hunk 3]
```diff
@@ -1,22 +1,22 @@
 {
-    "name": "@powsybl/diagram-viewer",
+    "name": "@powsybl/network-viewer",
     "version": "0.5.6",
     "description": "TypeScript library to integrate a PowSyBl SVG diagram in a JavaScript project.",
```
Comment (etiennehomer) [inline package.json:4]:
```suggestion
    "description": "TypeScript library to integrate PowSyBl network viewers in a JavaScript project.",
```

**Review by etiennehomer — CHANGES_REQUESTED** [review]
We keep powsybl-diagram-viewer as sonar project key ? We change it ?

File: README.md [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
+- network-map-viewer, to display the substations / voltage levels on a map
+
+The library is built with the Vite bundler.
 Node from v18+ is required to build with Vite.
 
 Installation using npm:  
-'npm install @powsybl/diagram-viewer'
+'npm install @powsybl/network-viewer'
```
Comment (etiennehomer) [inline README.md:13]: Is there  a procedure to rename an npm package ? We should find a way to make projects using @powsybl/diagram-viewer move to @powsybl/network-viewer I guess

Comment (flo-dup) [reply README.md:13]: It seems that you can't do that. But you could provide users with a deprecated message to tell them to use the new name: https://stackoverflow.com/questions/28371669/renaming-a-published-npm-module

Comment (flo-dup) [reply README.md:13]: `npm deprecate my-project-name@"<= latest-version" "This project has been renamed. Install using new-project-name instead."`

**Comment by flo-dup** [comment]
> We keep powsybl-diagram-viewer as sonar project key ? We change it ?

Right, we'd better change it. I'll change it in sonar, and then here.

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**